### PR TITLE
Add OpenStack Keystone Role module

### PR DESCRIPTION
**Pull request by emonty**

### CODE REVIEW

File: cloud/openstack/os_keystone_role.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+requirements:
+    - "python >= 2.6"
+    - "shade"
+'''
+
+EXAMPLES = '''
+# Create a role
+- os_keystone_role: name=demo description="Demo Role"
+'''
+
```
Comment (omgjlk) [inline cloud/openstack/os_keystone_role.py:57]: Please document return data (role).

File: cloud/openstack/os_keystone_role.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+description:
+    - Manage OpenStack Identity Roles.
+options:
+   name:
+     description:
+        - Role Name
+     required: true
+   description:
+     description:
+        - Role Description
```
Comment (omgjlk) [inline cloud/openstack/os_keystone_role.py:40]: Update docs to reflect that description can be modified on an existing role.

**Comment by gregdek** [comment]
@emonty ping for update

**Comment by cjchang** [comment]
Hi Monty, 

I am trying this module but more likely I have missed something. For quick testing in my ursula, I copied the os_keystone_role.py to the following directories.
./src/ansible/v2/ansible/modules/core/cloud/openstack/os_keystone_role.py
./src/ansible/lib/ansible/modules/core/cloud/openstack/os_keystone_role.py

When tried it in the ursula, using demo code
- name: try os_keystone_role.py
  os_keystone_role: name=demo
  run_once: true

I got this error and I can see the code has name is required. Do you know where I missed config?

TASK: [heat | try os_keystone_role.py] ****************************************
Sunday 09 August 2015  15:13:43 +0000 (0:00:00.727)       0:00:37.878 *********
failed: [22.22.0.18] => {"failed": true}
msg: unsupported parameter for module: name

FATAL: all hosts have already failed -- aborting

**Comment by cjchang** [comment]
I passed the name error by changing the following 
    argument_spec = openstack_full_argument_spec(
    argument_spec = dict(
        name=dict(required=True),
        description=dict(required=False, default=None),
        state=dict(default='present', choices=['absent', 'present']),
        # Override endpoint_type default since this is an admin function
        endpoint_type=dict(
            default='admin', choices=['public', 'internal', 'admin'])
    ))
to 
    argument_spec = openstack_full_argument_spec(
        name=dict(required=False),
        description=dict(required=False, default=None),
        state=dict(default='present', choices=['absent', 'present']),
        # Override endpoint_type default since this is an admin function
        endpoint_type=dict(
            default='admin', choices=['public', 'internal', 'admin'])
    )

but run into the following
TASK: [heat | os_keystone_role description="Demo Role"] ***********************
Monday 10 August 2015  17:25:39 +0000 (0:00:00.689)       0:00:15.850 *********
failed: [22.22.0.18] => {"failed": true, "parsed": false}
OpenSSH_5.9p1 Debian-5ubuntu1.4, OpenSSL 1.0.1 14 Mar 2012
debug1: Reading configuration data /root/.ssh/config
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: auto-mux: Trying existing master
debug1: mux_client_request_session: master session id: 2
Traceback (most recent call last):
  File "<stdin>", line 1823, in <module>
  File "<stdin>", line 84, in main
AttributeError: 'OperatorCloud' object has no attribute 'get_role'

FATAL: all hosts have already failed -- aborting

is there a different shade release to support get_role?

**Comment by emonty** [comment]
@cjchang nope - sorry - the shade patch is still outstanding. That's next on my list of things to merge. Sorry for the confusion.

**Comment by gregdek** [comment]
Hey @emonty -- ping on this module. 

cc: @emonty @shrews @dguerri @juliakreger @j2sol @rcarrillocruz

**Comment by emonty** [comment]
Still waiting on a thing. thanks for the ping - hope to get this one soon

**Comment by Shrews** [comment]
This PR has been replaced by PR: https://github.com/ansible/ansible-modules-extras/pull/1520

Please close this one.

**Pull request closed without merging**